### PR TITLE
feat: add user id datalayer

### PR DESCRIPTION
**Pull request by Aylie-Chou**
This patch push user id datalayer for gtm analysis & separate gtm data to different environments.

### CODE REVIEW

File: src/client.js [hunk 1]
```diff
@@ -11,9 +11,29 @@ import hashLinkScroll from './utils/hash-link-scroll'
...
+const _ = {
+  get,
+}
 
 const logger = loggerFactory.getLogger()
 const releaseBranch = globalEnv.releaseBranch
+const tagManagerArgs = {
+  master: {
+    gtmId: 'GTM-PRMXBBN',
+    auth: '2pJC7GotZqWa7HtmIgSFIg',
```
Comment (taylrj) [inline src/client.js:25]: Could you share the idea about why do we need `auth`, `preview` and why do they are different corresponds to different environments (master, staging) ?

Comment (Aylie-Chou) [reply src/client.js:25]: The official doc about environments can be found [here](https://support.google.com/tagmanager/answer/6311518).

The main reason we need to include environments is to publish testing gtm version to testing environment, thus we can show test data on GA matrix.
I separate master & staging for corresponding to releaseBranch in codebase. They can be combined into a single testing environment.

How do you consider about it ?

Comment (taylrj) [reply src/client.js:25]: I second this idea.

We can separate master & staging as different environments.

**Comment by taylrj** [comment]
LGTM

**Pull request merged**